### PR TITLE
add flux framework connector

### LINKED ISSUE #87: Interest in adding Flux Framework
Hiya! 

I'm interested in adding FluxFramework as a connector: https://flux-framework.readthedocs.io/en/latest/quickstart.html and I noticed that for other connectors with queues, the interaction is always an ssh command -> to a command line. Flux has an official Python SDK, however I'm assuming your preference is to not add additional dependencies and rely on command line stuffs? This is definitely do-able, I just want to check first. 

I also have a design question about the Kubernetes connector (or how it works in general) but I'll open a separate issue for that. Thanks!

### PR DESCRIPTION
**Pull request by vsoch**
This is start of work to close #87 

- I developed by way of adding a containerized example workflow, which has instructions for a user to quickly build and interact with the connector.
- For testing, I didn't see how the other connectors are tested, so I added a workflow file that has Flux in a container, and it runs the same workflow that I provided as a demo example. This seems to hit all parts of the Flux code that I added.
- One issue might be that my container build needs to add the examples, but they were left out in the `.dockerignore`. If this doesn't work for other container builds, let's brainstorm a way to satisfy both needs.
- I grabbed an example fairly randomly, and not knowing how it's supposed to work, I just edited it to use `flux run` (which typically replaces `mpirun`) and then print the worker numbers / ids and exit.

And of course any feedback / requests for change please let me know! Today was the first time I'm using streamflow, and I'm not a flux expert either, but I was able to figure out the commands (I think) we need.

### CODE REVIEW

**Comment by GlassOfWhiskey** [comment]
Hi @vsoch , thanks for the PR.
No problem for the `.dockerignore` update, examples had been excluded just to save time when building Docker containers, but for now the content of the folder is not heavy, so it will probably have no effect at all.

In general, the PR looks great. Just two small fixes before merging:

- Please run `make pyupgrade`, `make codespell`, and `make format` to adjust code formatting.
- Please move the content of the `test-flux.yml` directly in the `ci-tests.yaml` file.

Other connectors are not tested very well up to now. I am still working on the `ConnectorWrapper` feature to be able to run X-over-Docker and X-over-DockerCompose workflows. Then, there will be possible to recreate Slurm, PBS, Flux, and even Kubernetes locally on the CI machine using DockerCompose and use them to test connectors in a more consistent way. For now, what you did is more than enough :)

**Comment by vsoch** [comment]
All set! Let me know if you'd like me to ping a core Flux maintainer for an additional review - still early here but can in a few hours!

**Comment by vsoch** [comment]
okay pushed again using a different version of black for formatting!

File: pyproject.toml
Comment (GlassOfWhiskey) [inline pyproject.toml:1]: This update should be reverted.

File: streamflow/parser.py
Comment (GlassOfWhiskey) [inline streamflow/parser.py:1]: This update should be reverted

File: streamflow/provenance/run_crate.py
Comment (GlassOfWhiskey) [inline streamflow/provenance/run_crate.py:1]: All these updates should be reverted

**Comment by GlassOfWhiskey** [comment]
Hi @vsoch ,
unfortunately something strange happened with your `codespell` execution. For some reason it did not stick with `pyproject.toml` settings. The result is that some `crate` words have been automatically changed into `create`, but this is not correct, as here `crate` refers to the [RO-Crate](https://www.researchobject.org/ro-crate/) format.
I marked the affected files with comments. Could you please restore them?

**Review by GlassOfWhiskey — CHANGES_REQUESTED** [review]

**Comment by vsoch** [comment]
That's strange indeed - I followed your instruction and strange that came out like that.

I manually fixed those changes - should be OK now?

**Comment by GlassOfWhiskey** [comment]
Yes perfect. Thank you.

**Comment by vsoch** [comment]
Thank you!

Aside from these https://streamflow.di.unito.it/applications/ are there listings of apps (with code to share) that can be run with streamflow?

**Pull request merged**
